sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -393,6 +399,72 @@ def capabilities(self):
         log.debug("Server capabilities: %s", server_capabilities)
         return server_capabilities
 
+    def m_text_document__code_action(self, **kwargs) -> list[CodeActionTypedDict]:
+        params: TextDocumentCodeActionTypedDict = cast(
+            TextDocumentCodeActionTypedDict, kwargs
+        )
+        # Sample params:
+        # {
+        #     "textDocument": {
+        #         "uri": "file:///x%3A/vscode-robot/local_test/robot_check/checkmy.robot"
+        #     },
+        #     "range": {
+        #         "start": {"line": 12, "character": 5},
+        #         "end": {"line": 12, "character": 5},
+        #     },
+        #     "context": {
+        #         "diagnostics": [
+        #             {
+        #                 "range": {
+        #                     "start": {"line": 12, "character": 4},
+        #                     "end": {"line": 12, "character": 8},
+        #                 },
+        #                 "message": "Undefined keyword: rara.",
+        #                 "severity": 1,
+        #                 "source": "robotframework",
+        #             }
+        #         ],
+        #         "triggerKind": 1,
+        #     },
+        # }
+        code_action_list: list[CodeActionTypedDict] = []
+        diagnostics = params.get("context", {}).get("diagnostics", [])
+
+        if not diagnostics:
+            return code_action_list
+
+        incomplete_package_diags = [
+            d
+            for d in diagnostics
+            if d.get("code")
+            in [
+                ErrorCode.action_package_info_unsynchronized.value,
+                ErrorCode.agent_package_incomplete.value,
+            ]
+        ]
+
+        document_dir = Path(unquote(urlparse(params["textDocument"]["uri"]).path))
```

Comment:
What you should use here is:

`Path(sema4ai_ls_core.uris.to_fs_path(action_package_uri))`